specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2021-05-13-preview/desktopvirtualization.json

```diff
@@ -3637,181 +3966,496 @@
+          "$ref": "#/definitions/CredentialsProperties",
+          "description": "Credentials needed to create the virtual machine."
+        }
+      }
+    },
+    "CredentialsProperties": {
+      "description": "Credentials needed to create the virtual machine.",
+      "type": "object",
+      "properties": {
+        "localAdmin": {
+          "$ref": "#/definitions/KeyvaultCredentialProperties",
+          "description": "The local admin credentials."
+        },
+        "domainAdmin": {
+          "$ref": "#/definitions/KeyvaultCredentialProperties",
+          "description": "The domain admin credentials."
+        }
+      }
+    },
+    "KeyvaultCredentialProperties": {
+      "description": "Credentials kept in the keyvault.",
+      "type": "object",
+      "properties": {
+        "userName": {
+          "type": "string",
+          "description": "The user name to the account."
+        },
+        "passwordKeyVaultResourceId": {
+          "type": "string",
+          "description": "The keyvault resource id to the keyvault secrets."
+        },
+        "passwordSecretName": {
+          "type": "string",
+          "description": "The keyvault secret name the password is stored in."
+        }
+      }
+    },
+    "HostPoolUpdateParameters": {
+      "description": "Represents properties for a hostpool update.",
+      "type": "object",
+      "properties": {
+        "saveOriginalDisk": {
+          "type": "boolean",
+          "description": "Whether to save original disk. False by default."
+        },
+        "maxVMsRemovedDuringUpdate": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The maximum virtual machines to be removed during hostpool update."
+        },
+        "maintenanceAlerts": {
+          "type": "array",
+          "description": "The alerts given to customers for hostpool update.",
+          "items": {
+            "$ref": "#/definitions/MaintenanceAlertsProperties"
+          }
+        },
+        "logOffDelaySeconds": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Grace period before logging off users in seconds."
+        },
+        "logOffMessage": {
+          "type": "string",
+          "description": "Log off message sent to user for logoff."
+        },
+        "scheduledTime": {
+          "$ref": "#/definitions/ScheduledTimeProperties",
+          "description": "When set schedules the hostpool update at specific time."
+        }
+      }
+    },
+    "MaintenanceAlertsProperties": {
+      "type": "object",
+      "description": "Properties for host pool update maintenance alerts.",
+      "properties": {
+        "message": {
+          "type": "string",
+          "description": "The path to the legacy object to migrate."
+        },
+        "duration": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The duration of the hostpool update in seconds."
```

Comment:
should it be minutes instead?

Reply:
@Shoeliee

Reply:
seconds - the admin can opt to send out a maintenance notifications at intervals of seconds.

Reply:
For maintenance alert duration field, Scott told me that it is only allowed to provide up to minute from portal. And it is not necessary to support in seconds for maintenance alert. would you please help me to understand why customers need to control down to second?

Reply:
Vidya is OK with minutes as the unit

Reply:
log off alert - 60 mins is the max that can be configured
maintenance alerts
before hp update kicks off - is between 1 and 5 days
after hp update kicks off - should be > log off alerts and <= 120 mins

Reply:
Thanks for using minutes as the unit. @datr93 please help to update.